### PR TITLE
Rename the GIDGoogleUser method `doWithFreshTokens:`

### PR DESCRIPTION
**Pull request by Alex-4-Git**
The objc API method `doWithFreshTokens` translates into `do` in Swift which is not understandable well.
In this PR we change this to `refreshTokensIfNeededWithCompletion` so in Swift client can use it as `refreshTokensIfNeeded(completion:)`

### CODE REVIEW

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h [hunk 1]
```diff
@@ -76,14 +76,14 @@ NS_ASSUME_NONNULL_BEGIN
...
 ///     refresh tokens was unsuccessful.  The block will be called asynchronously on the main queue.
-- (void)doWithFreshTokens:(void (^)(GIDGoogleUser *_Nullable user,
-                                    NSError *_Nullable error))completion;
+- (void)refreshTokensWithCompletion:(void (^)(GIDGoogleUser *_Nullable user,
+                                              NSError *_Nullable error))completion;
 
 #if TARGET_OS_IOS || TARGET_OS_MACCATALYST
 
-/// Starts an interactive consent flow on iOS to add scopes to the user's grants.
+/// Starts an interactive consent flow on iOS to add scopes to the current user's grants.
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:84]: Was this reverted unintentionally?

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h [hunk 2]
```diff
@@ -76,14 +76,14 @@ NS_ASSUME_NONNULL_BEGIN
...
 ///
-/// The completion will be called at the end of this process.  If successful, a `GIDUserAuth`
+/// The completion will be called at the end of this process.  If successful, a new `GIDGoogleUser`
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:86]: Same as above.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h [hunk 3]
```diff
@@ -100,19 +100,19 @@ NS_ASSUME_NONNULL_BEGIN
 
 #elif TARGET_OS_OSX
 
-/// Starts an interactive consent flow on macOS to add scopes to the user's grants.
+/// Starts an interactive consent flow on macOS to add scopes to the current user's grants.
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:103]: Same as above.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h [hunk 4]
```diff
@@ -100,19 +100,19 @@ NS_ASSUME_NONNULL_BEGIN
...
 ///
-/// The completion will be called at the end of this process.  If successful, a `GIDUserAuth`
+/// The completion will be called at the end of this process.  If successful, a new `GIDGoogleUser`
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:105]: Same as above.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h [hunk 5]
```diff
@@ -100,19 +100,19 @@ NS_ASSUME_NONNULL_BEGIN
...
 ///
 /// @param scopes An array of scopes to ask the user to consent to.
 /// @param presentingWindow The window used to supply `presentationContextProvider` for `ASWebAuthenticationSession`.
 /// @param completion The block that is called on completion.  This block will be called asynchronously
 ///     on the main queue.
 - (void)addScopes:(NSArray<NSString *> *)scopes
-    presentingWindow:(NSWindow *)presentingWindow
-          completion:(nullable void (^)(GIDUserAuth *_Nullable userAuth,
-                                        NSError *_Nullable error))completion;
+ presentingWindow:(NSWindow *)presentingWindow
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:113]: Same as above.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h [hunk 6]
```diff
@@ -76,14 +76,14 @@ NS_ASSUME_NONNULL_BEGIN
 ///
 /// @param completion A completion block that takes a `GIDGoogleUser` or an error if the attempt to
 ///     refresh tokens was unsuccessful.  The block will be called asynchronously on the main queue.
-- (void)doWithFreshTokens:(void (^)(GIDGoogleUser *_Nullable user,
-                                    NSError *_Nullable error))completion;
+- (void)refreshTokensWithCompletion:(void (^)(GIDGoogleUser *_Nullable user,
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:79]: We may want to indicate (via the method name) that a token refresh will only be performed when needed.

Comment (Alex-4-Git) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:79]: `refreshTokensIfNeddedWithCompletion`? Do you have a better name on your mind?

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h (hunk 1 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:84]: Intended.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h (hunk 2 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h]: Fixed.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h (hunk 3 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:103]: Intended.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h (hunk 4 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h]: Fixed.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h (hunk 5 above)
Comment (Alex-4-Git) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h]: Fixed.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h (hunk 6 above)
Comment (petea) [reply GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:79]: I think that works.  We'll have a chance to review our naming before release.

File: GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h [hunk 7]
```diff
@@ -76,14 +76,14 @@ NS_ASSUME_NONNULL_BEGIN
...
+                                              NSError *_Nullable error))completion;
 
 #if TARGET_OS_IOS || TARGET_OS_MACCATALYST
 
-/// Starts an interactive consent flow on iOS to add scopes to the user's grants.
+/// Starts an interactive consent flow on iOS to add scopes to the current user's grants.
 ///
-/// The completion will be called at the end of this process.  If successful, a `GIDUserAuth`
+/// The completion will be called at the end of this process.  If successful, a `GIDUserAuth``
```
Comment (petea) [inline GoogleSignIn/Sources/Public/GoogleSignIn/GIDGoogleUser.h:86]: Typo?

**Review by petea — APPROVED** [review]

**Review by mdmathias — APPROVED** [review]

**Pull request merged**